src/control-plane/backend/lambda/api/test/api/sql-builder.test.ts

```diff
@@ -3394,191 +3833,232 @@ describe('SQL Builder test', () => {
-            session_id
-          ORDER BY
-            event_timestamp asc
-        ) as step_1,
-        ROW_NUMBER() OVER (
-          PARTITION BY
-            user_pseudo_id,
-            session_id
-          ORDER BY
-            event_timestamp asc
-        ) + 1 as step_2
-      from
-        (
-          select
-            event_date,
-            event_name,
-            user_pseudo_id,
-            event_id,
-            event_timestamp,
-            session_id,
-            replace(node, '"', '') as node
-          from
-            mid_table
-        ) t
-    )
-  select
-    a.event_date as event_date,
-    a.node || '_' || a.step_1 as source,
-    CASE
-      WHEN b.node is not null THEN b.node || '_' || a.step_2
-      ELSE 'other_' || a.step_2
-    END as target,
-    count(distinct a.user_pseudo_id) as weight
-  from
-    data a
-    left join data b on a.user_pseudo_id = b.user_pseudo_id
-    and a.session_id = b.session_id
-    and a.step_2 = b.step_1
-  where
-    a.step_2 <= 10
-  group by
-    a.event_date,
-    a.node || '_' || a.step_1,
-    CASE
-      WHEN b.node is not null THEN b.node || '_' || a.step_2
-      ELSE 'other_' || a.step_2
-    END`;
+            ),
+            'YYYY-MM-DD'
+          ) || ' - ' || TO_CHAR(
+            date_trunc(
+              'week',
+              (
+                TIMESTAMP 'epoch' + cast(event_timestamp / 1000 as bigint) * INTERVAL '1 second'
+              ) + INTERVAL '6 days'
+            ),
+            'YYYY-MM-DD'
+          ) as week,
+          TO_CHAR(
+            TIMESTAMP 'epoch' + cast(event_timestamp / 1000 as bigint) * INTERVAL '1 second',
+            'YYYY-MM-DD'
+          ) as day,
+          TO_CHAR(
+            TIMESTAMP 'epoch' + cast(event_timestamp / 1000 as bigint) * INTERVAL '1 second',
+            'YYYY-MM-DD HH24'
+          ) || '00:00' as hour,
+          event_params,
+          user_properties,
+          event_date,
+          event_name,
+          event_id,
+          event_bundle_sequence_id::bigint as event_bundle_sequence_id,
+          event_previous_timestamp::bigint as event_previous_timestamp,
+          event_server_timestamp_offset::bigint as event_server_timestamp_offset,
+          event_timestamp::bigint as event_timestamp,
+          ingest_timestamp,
+          event_value_in_usd,
+          app_info.app_id::varchar as app_info_app_id,
+          app_info.id::varchar as app_info_package_id,
+          app_info.install_source::varchar as app_info_install_source,
+          app_info.version::varchar as app_info_version,
+          device.vendor_id::varchar as device_id,
+          device.mobile_brand_name::varchar as device_mobile_brand_name,
+          device.mobile_model_name::varchar as device_mobile_model_name,
+          device.manufacturer::varchar as device_manufacturer,
+          device.screen_width::bigint as device_screen_width,
+          device.screen_height::bigint as device_screen_height,
```

Comment:
Add tests for the common attribute `device.viewport_width` and `device.viewport_height` for the Web SDK